Core/Core/Login/LoginWebPresenter.swift

```diff
@@ -117,30 +117,24 @@ class LoginWebPresenter {
             let mobileVerify = mobileVerifyModel, let baseURL = mobileVerify.base_url {
             task?.cancel()
             task = URLSessionAPI(urlSession: session).makeRequest(PostLoginOAuthRequest(client: mobileVerify, code: code)) { [weak self] (response, _, error) in
-                if let model = response {
-                    DispatchQueue.main.async {
-                        self?.view?.show(LoadingViewController.create(), sender: nil)
-                        self?.loginDelegate?.userDidLogin(session: LoginSession(
-                            accessToken: model.access_token,
-                            baseURL: baseURL,
-                            expiresAt: model.expires_in.flatMap { Date().addingTimeInterval($0) },
-                            locale: model.user.effective_locale,
-                            refreshToken: model.refresh_token,
-                            userID: model.user.id.value,
-                            userName: model.user.name,
-                            userEmail: model.user.email,
-                            clientID: mobileVerify.client_id,
-                            clientSecret: mobileVerify.client_secret
-                        ))
-                    }
-                }
-                if let error = error {
-                    DispatchQueue.main.async {
-                        self?.view?.showError(error)
+                DispatchQueue.main.async {
+                    guard let token = response, error == nil else {
+                        self?.view?.showError(error ?? NSError.internalError())
+                        return
                     }
+                    let session = LoginSession(
+                        accessToken: token.access_token,
+                        baseURL: baseURL,
+                        refreshToken: token.refresh_token,
```

Comment:
Do we not get an `expires_in`? That could be useful for proactively keeping an active token.

Reply:
On it.